contracts/mirror_staking/src/staking.rs

```diff
@@ -148,6 +152,134 @@ pub fn decrease_short_token<S: Storage, A: Api, Q: Querier>(
     })
 }
 
+pub fn auto_stake<S: Storage, A: Api, Q: Querier>(
+    deps: &mut Extern<S, A, Q>,
+    env: Env,
+    assets: [Asset; 2],
+    slippage_tolerance: Option<Decimal>,
+) -> HandleResult {
+    let config: Config = read_config(&deps.storage)?;
+    let terraswap_factory: HumanAddr = deps.api.human_address(&config.terraswap_factory)?;
+
+    let mut native_asset_op: Option<Asset> = None;
+    let mut token_info_op: Option<(HumanAddr, Uint128)> = None;
+    for asset in assets.iter() {
+        match asset.info.clone() {
+            AssetInfo::NativeToken { .. } => {
+                asset.assert_sent_native_token_balance(&env)?;
+                native_asset_op = Some(asset.clone())
+            }
+            AssetInfo::Token { contract_addr } => {
+                token_info_op = Some((contract_addr, asset.amount))
+            }
+        }
+    }
+
+    // will fail if one of them is missing
+    let native_asset: Asset = match native_asset_op {
+        Some(v) => v,
+        None => return Err(StdError::generic_err("Missing native asset")),
+    };
+    let (token_addr, token_amount) = match token_info_op {
```

Comment:
maybe good to check the token is in the asset pool store

Reply:
I had that check in the AutStakeHook, but moved it to AutoStake operation, I agree it makes more sense to check before.